src/App.tsx

```diff
@@ -1,26 +1,110 @@
+import { handleDeleteTodos } from './utils/handleDeleteTodos';
+
+import { Header, TodoList, Footer, ErrorMessage, TodoItem } from './components';
+import { handleUpdateTodos } from './utils/handleUpdateTodos';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoData, setNewTodoData] = useState<Partial<Todo>>({});
+
+  const [idsForDelete, setIdsForDelete] = useState<number[]>([]);
+  const [idsForUpdate, setIdsForUpdate] = useState<number[]>([]);
+
+  const [error, setError] = useState<Errors>(Errors.DEFAULT);
+
+  const [selectedFilter, setSelectedFilter] = useState<FilterBy>(FilterBy.ALL);
+
+  const completedTodosId = useMemo(() => {
+    return getFilteredTodos(todos, FilterBy.COMPLETED).map(todo => todo.id);
+  }, [todos]);
+
+  const numberOfActiveTodos = useMemo(() => {
+    return getFilteredTodos(todos, FilterBy.ACTIVE).length;
+  }, [todos]);
+
+  const filteredTodos = useMemo(() => {
+    return getFilteredTodos(todos, selectedFilter);
+  }, [todos, selectedFilter]);
+
+  useEffect(() => {
+    if (idsForDelete.length) {
+      handleDeleteTodos(idsForDelete, setTodos, setIdsForDelete, setError);
+    }
+  }, [idsForDelete]);
+
+  useEffect(() => {
+    if (idsForUpdate.length) {
+      handleUpdateTodos(
+        idsForUpdate,
+        newTodoData,
+        setTodos,
+        setIdsForUpdate,
+        setError,
+      );
+    }
+  }, [idsForUpdate, newTodoData]);
+
+  useEffect(() => {
+    handleFetchTodos(setTodos, setError);
+  }, []);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          tempTodo={tempTodo}
+          numberOfActiveTodos={numberOfActiveTodos}
+          setTodos={setTodos}
+          setError={setError}
+          setTempTodo={setTempTodo}
+          setIdsForUpdate={setIdsForUpdate}
+          setNewTodoData={setNewTodoData}
+        />
+
+        <TodoList
+          todos={filteredTodos}
+          idsForDelete={idsForDelete}
+          idsForUpdate={idsForUpdate}
+          setIdsForDelete={setIdsForDelete}
+          setIdsForUpdate={setIdsForUpdate}
+          setNewTodoData={setNewTodoData}
+        />
+
+        {tempTodo && (
+          <TodoItem
+            todo={tempTodo}
+            setIdsForDelete={setIdsForDelete}
+            setIdsForUpdate={setIdsForUpdate}
+            setNewTodoData={setNewTodoData}
+          />
+        )}
```

Comment:
I guess it can be put inside TodoList component

Reply:
Its makes according to part 2 of the todo app
![image](https://github.com/user-attachments/assets/950168f5-5a2a-4be1-b6ad-abcf36e86561)